marxs/optics/multiLayerMirror.py

```diff
@@ -0,0 +1,104 @@
+        # eliminate photons that do not reach the mirror
+        doesIntersect, intersection, local = self.intersect(photons['dir'], photons['pos'])
+        photons['probability'] *= doesIntersect
+        
+        # split polarization into parallel and other components
+        beam_dir = (photons['dir'][:,0:3]).copy()
+        for i in range(0, len(beam_dir)):
+        	beam_dir[i] /= np.linalg.norm(beam_dir[i])
+        # v_1 is parallel to plane, v_2 is third direction remaining
+        v_1 = np.cross(beam_dir, self.geometry['e_x'][0:3])
+        v_2 = np.cross(beam_dir, v_1)
+        # find polarization in each direction
+        polarization = (photons['polarization'][:,0:3]).copy()
+        p_v_1 = np.zeros(len(polarization))
+        p_v_2 = np.zeros(len(polarization))
+        for i in range(0, len(polarization)):
+        	v_1[i] /= np.linalg.norm(v_1[i])
+        	v_2[i] /= np.linalg.norm(v_2[i])
+        	p_v_1[i] = np.dot(polarization[i], v_1[i])   # just the cosine between the vectors
+        	p_v_2[i] = np.dot(polarization[i], v_2[i])
+        
+        # reflect (change direction)
+        directions = photons['dir']
+        pos4d_inv = np.linalg.inv(self.pos4d)
+        directions = directions.T
+        directions = np.dot(pos4d_inv, directions)
+        directions[0,:] *= -1
+        directions = np.dot(self.pos4d, directions)
+        directions = directions.T
+        photons['dir'] = directions
+        
+        # adjust polarization vectors after reflection
+        # find new v_2
+        new_beam_dir = (photons['dir'][:,0:3]).copy()
+        for i in range(0, len(new_beam_dir)):
+        	new_beam_dir[i] /= np.linalg.norm(new_beam_dir[i])
+        new_v_2 = np.cross(new_beam_dir, v_1)
+        for i in range(0, len(polarization)):
+        	photons['polarization'][i,0:3] = p_v_1[i] * v_1[i] + p_v_2[i] * new_v_2[i]
+        
+        # set position to intersection point
+        photons['pos'] = intersection
+        
+        # read in correct reflecting probability file, now in table format
+    	reflectFile = ascii.read(self.fileName)
+        
+        # find reflectivity adjustment due to polarization in testing
+        polarizedFile = ascii.read('./marxs/optics/docs/ALSpolarization2.txt')
+        tested_polarized_fraction = np.interp(photons['energy'], polarizedFile['Photon energy'] / 1000, polarizedFile['Polarization'])
+        
+        # find probability of being reflected due to position
+        local_intersection = h2e((np.dot(pos4d_inv, intersection.T)).T)
+        local_coords_in_file = reflectFile['X(mm)'] / np.linalg.norm(self.geometry['v_y']) - 1
+        peak_wavelength = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['Peak lambda'])
+        max_refl = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['Peak']) / tested_polarized_fraction
+        spread_refl = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['FWHM(nm)'])
+        
```

Comment:
I can't really check this, because I don't know the files, so I just trust you here - after all, that's why we have tests.